great_expectations/expectations/expectation.py

```diff
@@ -75,6 +75,7 @@
 from great_expectations.expectations.sql_tokens_and_types import (
     valid_sql_tokens_and_types,
 )
+from great_expectations.expectations.window import Window  # noqa: TCH001
```

Comment:
Instead of ignoring TCH001, you can move this to the type checking block.

Reply:
This was causing a pydantic inheritance issue, I'll repro and put error here

Reply:
Oh, yes. Pydantic and static type checking are sometimes at odds since pydantic is very dynamic by nature. It could by mypy thinks this is only used in type checking but pydantic uses that info at runtime for validations.

Reply:
With `Window` import in type checking block, ruff is happy but test fails with:
```
 if validation_error:
>           raise validation_error
E           pydantic.v1.error_wrappers.ValidationError: 1 validation error for ExpectColumnValuesToBeInSet
E           windows
E             field required (type=value_error.missing)
```

Reply:
@Kilo59 we talked about a fix for base classes today at BE CoP, but this seems like a different issue, because its not a base class, its import of a model for typing. Any ideas?

Reply:
It's still essentially the same issue. Did you try putting adding this model (or its parent) to the ruff list of runtime evaluated base classes?

Reply:
Adding it to runtime evaluated base classes didn't fix it, I'll take another look at it tomorrow

Reply:
Can you commit and push your attempted fix (tomorrow)?

It could be that we need the gx pydantic compatibility class in here, or there could be some issue with Metclasses, but Fluent datasources use metaclasses too and so I doubt that's it. 
https://github.com/great-expectations/great_expectations/blob/dec5dcede7a562659af2d62977e36d9596ea2848/pyproject.toml#L394-L401

Reply:
Forced commit 7eb605a41. I think its not working as expect because my Window class is not a base class, its just an import.

Here is a [demo](https://gxlabs.zoom.us/clips/share/zYsDKPfxkhYg5jKnq65t6V04cY0WPYX7feuMiN0Q31BdZuEJwQCg2jrGXqbOX0jJ0lblW_RWgsj1lwCWoOXn1zew.od5ECZ0pAnpaWrQ4) of the errors

Reply:
The class that needs to be recognized as needing runtime annotations is `Expectation` not `Window`.

Made a suggestion here.
https://github.com/great-expectations/great_expectations/pull/10402#discussion_r1767102792

Reply:
Ah I see I had the wrong class. I also applied your suggestion, which worked, then I trimmed it down to one net additional class:

 ```
great_expectations.compatibility.pydantic.BaseModel
```
47a6bbf39

All good now - thank you for the support